### PR TITLE
feat(room): check and add methods for room-svc

### PR DESCRIPTION
**Pull request by shivadharmi**
closes https://github.com/helpwave/services/issues/147

### CODE REVIEW

File: services/task-svc/internal/room/models/models.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ type Room struct {
 	ID             uuid.UUID `gorm:"column:id"`
 	OrganizationID uuid.UUID `gorm:"column:organization_id"`
 	Beds           []Bed     `gorm:"foreignKey:RoomID"`
+	WardID         uuid.UUID `gorm:"column:ward_id"`
```
Comment (MaxSchaefer) [inline services/task-svc/internal/room/models/models.go:14]: We should use the `Ward` as an entity.

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

**Comment by MaxSchaefer** [comment]
Hey @shivadharmi, are you keeping track of the requested changes, or can I help you with that?

File: services/task-svc/internal/room/room.go [hunk 2]
```diff
@@ -97,3 +114,105 @@ func (ServiceServer) UpdateRoom(ctx context.Context, req *pb.UpdateRoomRequest)
...
+
+	return &pb.GetRoomsResponse{
+		Rooms: roomsResponse,
+	}, nil
+}
+
+func (ServiceServer) GetRoomsByWard(ctx context.Context, req *pb.GetRoomsByWardRequest) (*pb.GetRoomsByWardResponse, error) {
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
```
Comment (use-to) [inline services/task-svc/internal/room/room.go:155]: Authorization

File: services/task-svc/internal/room/room.go [hunk 3]
```diff
@@ -97,3 +114,105 @@ func (ServiceServer) UpdateRoom(ctx context.Context, req *pb.UpdateRoomRequest)
...
+	}
+
+	room := models.Room{ID: id}
+
+	if err := db.Delete(&room).Error; err != nil {
+		log.Warn().Err(err).Msg("database error")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// Do we need to delete the beds when we delete room?
```
Comment (use-to) [inline services/task-svc/internal/room/room.go:208]: Yes, this is required.

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**